iocore/net/quic/qlog/QLog.h

```diff
@@ -46,9 +46,13 @@ class Trace
     VantagePointType flow = VantagePointType::unknown;
   };
 
-  Trace(std::string odcid, std::string title = "", std::string desc = "") : _reference_time(Thread::get_hrtime()), _odcid(odcid) {}
+  Trace(const std::string &odcid, const std::string &title = "", const std::string &desc = "")
```

Comment:
When you use the defaults for the last 2 parameters, the compiler will have to generate temporary strings and pass their addresses, like in this example:  https://godbolt.org/z/s976ce .  Surprisingly, the optimizer is not smart enough to reuse the temporary, even within the small function g() .  So you may want to create your own named empty string as the default value for the parameter, similar to S::Default in the example.

Reply:
I appreciate the feedback but it doesn't seem to be related to const-ify. Please file an issue or make a PR.

Reply:
Well yes and no.  With the original code, you create a new string for each parameter whether or not you use the default parameter value.  So you want to avoid this even when you use the default value.